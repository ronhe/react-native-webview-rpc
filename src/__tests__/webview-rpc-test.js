/* global jest, test, expect */

import React from 'react';
import renderer from 'react-test-renderer';

import WebViewRpc from '../native/webview-rpc';


jest.mock('WebView', () => 'WebView');


test('renders correctly', () => {
  const tree = renderer.create(<WebViewRpc />).toJSON();
  // tree.props.injectedJavaScript = tree.props.injectedJavaScript.replace(
  //   /rn-webview-rpc@\d+\.\d+\.\d+/,
  //   'rn-webview-rpc',
  // );
  expect(tree).toMatchSnapshot();
});
