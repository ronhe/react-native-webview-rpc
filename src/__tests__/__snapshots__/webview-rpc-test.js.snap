// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <RNCWKWebView
    cacheEnabled={true}
    injectedJavaScript=""
    messagingEnabled={true}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    source={null}
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
    useSharedProcessPool={true}
  />
</View>
`;
